contracts/layer-2/NFTBurner.sol

```diff
@@ -0,0 +1,163 @@
+// SPDX-License-Identifier: MIT
+// Unagi Contracts v1.0.0 (NFTBurner.sol)
+pragma solidity 0.8.12;
+
+import "@openzeppelin/contracts/interfaces/IERC721.sol";
+import "@openzeppelin/contracts/interfaces/IERC721Receiver.sol";
+import "@openzeppelin/contracts/interfaces/IERC721Receiver.sol";
+import "@openzeppelin/contracts/interfaces/IERC20.sol";
+import "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import "@openzeppelin/contracts/access/AccessControl.sol";
+import "solidity-bytes-utils/contracts/BytesLib.sol";
+
+/**
+ * @title NFTBurner
+ * @dev TODO
+ */
+contract NFTBurner is IERC721Receiver, AccessControl {
+    using SafeERC20 for IERC20;
+
+    address public constant DEAD_ADDRESS =
+        0x000000000000000000000000000000000000dEaD;
+
+    bytes32 public constant OPERATOR_ROLE = keccak256("OPERATOR_ROLE");
+
+    // Possible states for an existing token burn
+    bytes32 public constant BURN_RESERVED = keccak256("BURN_RESERVED");
+    bytes32 public constant BURN_EXECUTED = keccak256("BURN_EXECUTED");
+    bytes32 public constant BURN_REVERTED = keccak256("BURN_REVERTED");
+
+    struct Burn {
+        address from;
+        uint256[] tokenIds;
+        bytes32 state;
+    }
```

Comment:
Even if we don't use it for now, let's allow to burn some CHAMPs as well ! Would be nice to allow the CHAMPs to go to a custom address